File: rust src/round_wave.rs
use std::time;

fn main() {

    let mut i: f64 = 0.0;
    
    loop {
        println!("");
	let mut h: i64 = 0;
	while h < (100_f64 * i.cos().abs()) as i64 {
	    print!(" ");
	    h += 1;
	}

	print!("*");

	i += 0.1;

	std::thread::sleep(time::Duration::from_millis(65));
    }
}

File: java/round_wave.java
import java.util.*;
import java.math.*;

class Main {
    public static void main(String[] args) throws InterruptedException  {

	double i = 0.0;
	
	while (true) {
	    for (int h = 0; h < 100 * Math.abs(Math.cos(i)); h++) {
		System.out.print(" ");
	    }

	    System.out.println("*");

	    i+=0.1;

	    Thread.sleep(65);
	}
    }
}
